src/ResourceManagement/Sql/SqlDatabaseOperationsImpl.cs

```diff
@@ -182,6 +182,41 @@ public ISqlDatabase Get(string name)
             }
         }
 
+        ///GENMHASH:48BA5938680939720DEA90858B8FC7E4:131DBD7FC4DF46861698DE59CE4FDCED
+        public async Task<Microsoft.Azure.Management.Sql.Fluent.ISqlDatabase> GetBySqlServerAsync(ISqlServer sqlServer, string name, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            try
+            {
+                var inner = await this.sqlServerManager.Inner.Databases
+                    .GetAsync(sqlServer.ResourceGroupName, sqlServer.Name, name, cancellationToken: cancellationToken);
+                return inner != null ? new SqlDatabaseImpl(inner.Name, (SqlServerImpl)sqlServer, inner, sqlServer.Manager) : null;
+            }
+            catch (CloudException ex) when (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+            catch (AggregateException ex) when ((ex.InnerExceptions[0] as CloudException).Response.StatusCode == System.Net.HttpStatusCode.NotFound)
```

Comment:
>.InnerExceptions[0] [](start = 51, length = 19)

this is kind of unsafe - what if AggregatedException will not have any inner exceptions?

Reply:
I opened https://github.com/Azure/azure-libraries-for-net/issues/233 since this needs to be also fixed in the "core" classes.